tests/Backend/Mixed/SharingTest.php

```diff
@@ -1200,6 +1231,274 @@ function (array $column) {
+    {
+        $columnId = $tableId . '.transid';
+        $this->prepareTestMetadata($bucketId, $tableId, $columnId);
+
+        $response = $this->_client2->getSharedBucketDetail(
+            $project['id'],
+            $bucketId,
+            [
+                'include' => 'columns'
+            ]
+        );
+
+        $this->assertArrayHasKey('id', $response);
+        $this->assertArrayHasKey('description', $response);
+        $this->assertArrayHasKey('project', $response);
+        $this->assertArrayHasKey('tables', $response);
+        $this->assertArrayHasKey('created', $response);
+        $this->assertArrayHasKey('lastChangeDate', $response);
+        $this->assertArrayHasKey('dataSizeBytes', $response);
+        $this->assertArrayHasKey('rowsCount', $response);
+        $this->assertArrayHasKey('backend', $response);
+
+        $this->assertArrayHasKey('id', $response['project']);
+        $this->assertArrayHasKey('name', $response['project']);
+
+        $this->assertSame($bucketId, $response['id']);
+        $this->assertEquals($response['project']['id'], $project['id']);
+        $this->assertEquals($response['project']['name'], $project['name']);
+
+        $this->assertArrayHasKey('tables', $response);
+
+        $tableColumns = $this->_client->getTable($tableId)['columns'];
+        foreach ($response['tables'] as $table) {
+            $this->assertArrayNotHasKey('metadata', $table);
+            $this->assertArrayNotHasKey('columnMetadata', $table);
+            $this->assertArrayHasKey('columns', $table);
+            $this->assertCount(5, $table['columns']);
+            foreach ($table['columns'] as $column) {
+                $this->assertTrue(in_array($column, $tableColumns));
+            }
+        }
+
+        $response = $this->_client2->getSharedBucketDetail(
+            $project['id'],
+            $bucketId,
+            [
+                'include' => 'metadata,columns,tableMetadata,columnMetadata'
+            ]
+        );
+
+        $this->assertCount(1, $response['metadata']);
+
+        $sharedBucketMetadata = reset($response['metadata']);
+
+        $this->assertArrayHasKey('id', $sharedBucketMetadata);
+        $this->assertArrayHasKey('key', $sharedBucketMetadata);
+        $this->assertArrayHasKey('value', $sharedBucketMetadata);
+        $this->assertArrayHasKey('provider', $sharedBucketMetadata);
+        $this->assertArrayHasKey('timestamp', $sharedBucketMetadata);
+
+        $this->assertEquals('test', $sharedBucketMetadata['provider']);
+        $this->assertEquals('test.metadata.key', $sharedBucketMetadata['key']);
+        $this->assertEquals('test.metadata.value', $sharedBucketMetadata['value']);
+
+        foreach ($response['tables'] as $table) {
+            $this->assertArrayHasKey('metadata', $table);
+            $this->assertArrayHasKey('columns', $table);
+            $this->assertArrayHasKey('columnMetadata', $table);
+
+            $tableMetadata = reset($table['metadata']);
```

Comment:
všude bych přidal ještě assert na množství položek v těch metadatech - když třeba začneme nějak dědit nebo dynamicky vytvářet metadata tak aby to tady vyskočilo, že je to rozbité/jinak.

Reply:
(tj. v bucketu tabulce i column)

Reply:
A místo resetu bych dal to co očekáváš, že to bude - tím rovnou otestuješ i to, že se to vrací správnému column

Reply:
tj. $table['metadata']['id']

Reply:
(taktéž platí všude)

Reply:
^ sorry, překouknul jsem se - už to tak děláš s tím klíčem...

Reply:
Takže jen assertnout počet metadata

Reply:
Upravil som